cardano-cli/src/Cardano/CLI/Options/Flag.hs

```diff
@@ -0,0 +1,99 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE OverloadedRecordDot #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE NoFieldSelectors #-}
+
+module Cardano.CLI.Options.Flag
+  ( Flag (..)
+  , FlagOptions (..)
+  , isDefault
+  , mkFlag
+  , parserFromFlags
+  )
+where
+
+import Cardano.CLI.Vary
+import Cardano.CLI.Vary qualified as Vary
+
+import Control.Applicative
+import Options.Applicative (Parser)
+import Options.Applicative qualified as Opt
+
+-- | Options for a flag that control how it is rendered and parsed.
+newtype FlagOptions = FlagOptions
+  { isDefault :: Bool
```

Comment:
Use a sum type here to avoid boolean blindness.

Reply:
This is probably okay because it's not data that should be passed around.  Code should pass `FlagOptions` around instead and always access via the field and having specified the field there isn't blindness.  If this changes we can introduce a sum type.

Reply:
I still don't want `Bool`s dotted around the library.

Reply:
The `Bool` has been removed.